temporal-sdk/src/main/java/io/temporal/activity/ActivityCancellationType.java

```diff
@@ -19,24 +19,34 @@
 
 package io.temporal.activity;
 
-/** Defines Activity Execution cancellation behavior. */
+import io.temporal.client.ActivityCompletionException;
+
+/**
+ * In case of an activity's call scope cancellation the corresponding activity stub call fails with a
+ * {@link io.temporal.failure.CanceledFailure}.
+ * The different modes of this behavior are available and specified by this enum.
+ */
 public enum ActivityCancellationType {
   /**
-   * Wait for the Activity Execution to confirm any requested cancellation. An Activity Execution
-   * must Heartbeat to receive a cancellation notification. This can block the cancellation for a
-   * long time if the Activity Execution doesn't Heartbeat or chooses to ignore the cancellation
-   * request.
+   * Wait for the Activity Execution to confirm any requested cancellation.
+   * An Activity Execution must Heartbeat to receive a cancellation notification through {@link ActivityCompletionException}.
+   * This can block the cancellation for Workflow Execution for a long time if the Activity Execution doesn't Heartbeat
+   * or chooses to ignore the cancellation request.
```

Comment:
```suggestion
   * This can block the cancellation of a Workflow Execution for a long time if the Activity Execution doesn't Heartbeat
   * or chooses to ignore the cancellation request.
```